LongWarOfTheChosen/Src/LW_Overhaul/Classes/UIOutpostManagement.uc

```diff
@@ -83,176 +94,347 @@ function OnRadioTowerUpgradeClicked(UIButton Button)
+	m_BorderPadding = 15;
+	m_ItemPadding = 10;
+	
+	// KDM : The normal UI has 2 columns : rebel name, and rebel job; the fancy UI has a rebel perks column in the middle.
+	if (m_bUseFancyVersion)
+	{
+		m_NameHeaderPct = 0.45f;
+		m_PerksHeaderPct = 0.25f;
+		m_JobHeaderPct = 0.3f;
+	}
+	else
+	{
+		m_NameHeaderPct = 0.7f;
+		m_JobHeaderPct = 0.3f;
+	}
 
-    // Region Info
-	RegionalInfo = Spawn(class'UIScrollingText', MainPanel).InitScrollingText('Outpost_RegionalInfo_LW', "", 500, 453, ListBG.Y + 10 + 36.75);
-	kTag = XGParamTag(`XEXPANDCONTEXT.FindTag("XGParam"));
-	RegionalAIState = class'XComGameState_WorldRegion_LWStrategyAI'.static.GetRegionalAI(Region);
-	if (!RegionalAIState.bLiberated)
+	NextX = 0;
+	NextY = 0;
+	AdviserIconSize = 64;
+	AdviserBorderPadding = 4;
+	ScrollbarPadding = 10;
+	
+	super.InitScreen(InitController, InitMovie, InitName);
+
+	// KDM : The Haven screen is centered horizontally and vertically when viewed via the strategy map.
+	// When viewed via the Avenger; however, its vertical position must be set manually in order to fit a fancy 3D background.
+	panelW = 1000;
+    panelH = 830;
+	if (class'Utilities_LW'.static.IsOnStrategyMap())
 	{
-		kTag.IntValue0 = RegionalAIState.LocalAlertLevel;
-		kTag.IntValue1 = RegionalAIState.LocalVigilanceLevel;
-		kTag.IntValue2 = RegionalAIState.LocalForceLevel;
-		strRegionalInfo = `XEXPAND.ExpandString(m_strRegionalInfo);
+		panelY = (Movie.UI_RES_Y / 2) - (panelH / 2);
 	}
 	else
 	{
-		strRegionalInfo = class'UIResistanceManagement_LW'.default.m_strLiberated;
+		panelY = 80;
+	}
+
+	// KDM : On the Avenger, both the Resistance overview screen and, this, Haven screen are placed within a 3D movie;
+	// consequently, a 3D camera needs to be specified.
+	if (!class'Utilities_LW'.static.IsOnStrategyMap())
+	{
+		class'UIUtilities'.static.DisplayUI3D(DisplayTag, CameraTag, 0, true);
 	}
-	RegionalInfo.SetHTMLText("<p align=\'RIGHT\'><font size=\'24\' color=\'#fef4cb\'>" $ strRegionalInfo $ "</font></p>");
+
+	// KDM : Container which will hold our UI components : it's invisible
+    MainPanel = Spawn(class 'UIPanel', self);
+	MainPanel.bAnimateOnInit = false;
+	MainPanel.bIsNavigable = false;
+    MainPanel.InitPanel('ListContainer');
+	MainPanel.SetPosition((Movie.UI_RES_X / 2) - (panelW / 2), panelY);
+    
+	// KDM : Background rectangle
+	ListBG = Spawn(class'UIBGBox', MainPanel);
+	ListBG.bAnimateOnInit = false;
+	ListBG.LibID = class'UIUtilities_Controls'.const.MC_X2Background;
+	ListBG.InitBG('ListBG', 0, 0, panelW, panelH);
+    
+	// KDM : Header (includes title, and region sub-title) 
+	ListTitle = Spawn(class'UIX2PanelHeader', MainPanel);
+	ListTitle.bAnimateOnInit = false;
+	ListTitle.bIsNavigable = false;
+	ListTitle.InitPanelHeader('TitleHeader', m_strTitle, Region.GetDisplayName());
+	ListTitle.SetPosition(m_BorderPadding, m_BorderPadding);
+	ListTitle.SetHeaderWidth(panelW - m_BorderPadding * 2);
+    
+    NextY = ListTitle.Y + ListTitle.Height;
```

Comment:
OK, rather than point out every instance, would you mind just replacing all spaces indent with tabs? At least in new code. If there aren't too many instances of spaces in the code that's left untouched, then I wouldn't mind that being fixed as well. But if such a change were to drown out all the substantive changes in this commit, then don't. Hope that's clear. Sadly it's a bit subjective on how much pure formatting change is too much, so if in doubt, just leave existing code as is.